Translate tutorial (#20)
blitz-js/nl.blitzjs.com: app/pages/docs/tutorial.mdx

sidebar_label: Tutorial
---

> In deze tutorial maken we samen een eenvoudige stemming-applicatie.

Reviewer: ```sugestion
In deze tutorial maken we samen een eenvoudige stemmingapplicatie.
```

Author: Ik vond het een moeilijke term om duidelijk te vertalen. Zou **stemapplicatie** misschien nog beter zijn?

Reviewer: Ik heb op wat websites i.v.m. verkiezingen gekeken omdat ik ook niet zeker was en er wordt altijd stemmingapplicatie gebruikt dus zal wel goed zijn.

└── yarn.lock
```

Deze bestanden zijn:

> - De `app/` map bevat het meeste van je project. Hier plaats je pagina's

Reviewer: ```suggestion
- De `app/` map bevat het grootste deel van je project. Hier plaats je pagina's
```

  of API-routes.

- De `app/pages/` map is de map met primaire pagina's. Als je Next.js hebt
  gebruikt, merk je meteen dat dit anders is. In Blitz kan je
  verschillende `pages` mappen hebben die op build time worden
  samengevoegd.

- De `app/core/` map is de belangrijkste plaats om componenten, hooks,
  etc. op te slaan die doorheen je applicatie worden gebruikt.

- `db/` is waar je databaseconfiguratie terecht komt. Als je modellen
  schrijft of migraties nakijkt, kan je hier terecht.

- `public/` is een map waarin je alle statische assets plaatst. Als je
  afbeeldingen, bestanden of video's hebt die je in je applicatie wilt
  gebruiken, kun je ze hier zetten.

- `.babelrc.js`, `.env`, etc. ("dotfiles") zijn configuratiebestanden voor

>   verschillende stukken JavaScript tooling.

Reviewer: ```suggestion
- `.babelrc.js`, `.env`, etc. ("dotfiles") zijn configuratiebestanden voor
  de verschillende onderdelen van JavaScript tooling.
```

event - compiled successfully
```

Nu de server draait, ga je via je webbrowser naar
[localhost:3000](http://localhost:3000). Je ziet een welkomstpagina met
het Blitz-logo. Het is gelukt!

## Registreer je als gebruiker {#sign-up-as-a-user}

> Blitz-apps worden gemaakt met gebruikersregistratie en login al ingesteld!

Reviewer: ```suggestion
Blitz apps worden gemaakt met gebruikersregistratie en login al ingesteld!
```

Author: De schrijfwijze met koppelteken wordt ook op andere pagina's al gebruikt, vandaar dat ik dit heb doorgetrokken.

Reviewer: Ja, ik denk dat ik ze ook door elkaar heb gegooid lol. Was ook niet meer zeker.

Dus laten we eens proberen. Klik op de knop **Sign Up**. Voer een
e-mailadres en wachtwoord in en klik op **Create Account**. Hierna word je
teruggeleid naar de startpagina waar je de `id` en `role` van je gebruiker
kan zien.

> Als je wil, kan je ook proberen uitloggen en terug inloggen. Of klik op

Reviewer: ```suggestion
Als je wil, kan je ook proberen uit te loggen en terug in te loggen. Of klik op
```

`blitz prisma studio` uitvoeren in de terminal om een web interface te
openen waar je de data in je database kan bekijken.

Wanneer je je eerste echte project start, zal je een meer schaalbare
database willen gebruiken zoals PostgreSQL, om te vermijden dat je
onderweg van database moet veranderen. Voor meer informatie, bekijk
[Database overzicht](database-overview). Voor nu gaan we verder met de
standaard SQLite database.

## Code voor onze modellen scaffolden {#scaffolding-code-for-our-models}

Blitz heeft een handig CLI commando genaamd [`generate`](./cli-generate)
om boilerplate code te scaffolden. We zullen `generate` gebruiken om twee
modellen aan te maken: `Question` en `Choice`. Een `Question` bevat de
tekst van de vraag en een lijst met keuzes. Een `Choice` bevat de tekst
van de keuze, het aantal stemmen, en een geassocieerde vraag. Blitz maakt
voor beide modellen automatisch een id, een _aangemaakt_ timestamp en een
_laatst bewerkt_ timestamp.

#### Eerst maken we alles voor het `Question` model:

```sh
blitz generate all question text:string
```

Druk op **Enter** als er wordt gevraagd om `prisma migrate` uit te voeren,

> die je databaseschema zal updaten met het nieuwe model. Er zal een naam

Reviewer: ```suggestion
dit zal je databaseschema updaten met het nieuwe model. Er zal een naam
```

> worden gebraagd, dus typ iets zoals "add question".

Reviewer: ```suggestion
worden gevraagd, dus typ iets zoals "add question".

```

> worden gebraagd, dus typ iets zoals "add question".

Reviewer: ```suggestion
Druk op **Enter** als er wordt gevraagd om `prisma migrate` uit te voeren,
dit zal je databaseschema updaten met het nieuwe model. Er zal een naam
worden gevraagd, dus typ iets zoals "add question".
```

> die je databaseschema zal updaten met het nieuwe model. Er zal een naam
> worden gebraagd, dus typ iets zoals "add question".

Reviewer: ```suggestion
dit zal je databaseschema updaten met het nieuwe model. Er zal een naam
worden gevraagd, dus typ iets zoals "add question".
```

> Druk op **Enter** als er wordt gevraagd om `prisma migrate` uit te voeren,
> die je databaseschema zal updaten met het nieuwe model. Er zal een naam
> worden gebraagd, dus typ iets zoals "add question".

Reviewer: ```suggestion
Druk op **Enter** als er wordt gevraagd om `prisma migrate` uit te voeren,
dit zal je databaseschema updaten met het nieuwe model. Er zal een naam
worden gevraagd, dus typ iets zoals "add question".
```

}
```

Voer dan `blitz prisma generate` uit om de Prisma client te updaten met

> het aangepaste schema. Er is geen database migratie nodig, omdat er geen

Reviewer: ```
het aangepaste schema. Er is geen databasemigratie nodig, omdat er geen

```

```sh
blitz console
```

Eens je in de console bent, kan je de database client verkennen:

```sh
# Er zitten nog geen vragen in het systeem.
⚡ > await db.question.findMany()
[]

# Maak een nieuwe vraag:
⚡ > let q = await db.question.create({data: {text: "Wat is er nieuw?"}})
undefined

# Bekijk het hele object:
⚡ > q
{
  id: 1,
  createdAt: 2020-06-15T15:06:14.959Z,
  updatedAt: 2020-06-15T15:06:14.959Z,
  text: "Wat is er nieuw?"
}

# Of, haal individuele waardes uit het object op:
⚡ > q.text
"Wat is er nieuw?"

# Verander waardes door de update functie te gebruiken:
⚡ > q = await db.question.update({where: {id: 1}, data: {text: "Hoe gaat het ermee?"}})
{
  id: 1,
  createdAt: 2020-06-15T15:06:14.959Z,
  updatedAt: 2020-06-15T15:13:17.394Z,
  text: "Hoe gaat het ermee?"
}

# db.question.findMany() toont nu alle vragen in de database:
⚡ > await db.question.findMany()
[
  {
    id: 1,
    createdAt: 2020-06-15T15:06:14.959Z,
    updatedAt: 2020-06-15T15:13:17.394Z,
    text: "Hoe gaat het ermee?"
  }
]
```

## Update de gegenereerde code voor de attributen van ons model {#update-generated-code-for-our-model-attributes}

<!-- prettier-ignore-start -->
:::info

> Voor we de app opnieuw runnen, moeten we wat van de code die gegenereerd is, aanpassen. Uiteindelijk zullen deze aanpassingen niet meer nodig zijn, maar voor nu moeten we nog rond een aantal openstaande problemen werken.

Reviewer: ```suggestion
Voor we de app opnieuw uitvoeren, moeten we een deel van de code die gegenereerd is, aanpassen. 
```

:::
<!-- prettier-ignore-end -->

> De gegenereerde paginainhoud maakt nog geen gebruik van de eigenlijke

Reviewer: ```suggestion
De gegenereerde pagina-inhoud maakt nog geen gebruik van de eigenlijke

``

// ...
```

### `deleteQuestion` mutatie {#delete-question-mutation}

[//]: # "Remove this section once Prisma supports cascading deletes"

Prisma ondersteunt nog geen "cascading deletes". In de context van deze

> tutorial betekent dat dat de `Choice` data niet wordt verwijderd wanneerd

Reviewer: ```suggestion
tutorial betekent dit dat de `Choice` data niet wordt verwijderd wanneer
```

)
```

#### Probeer het uit

Nu kan je naar `localhost:3000/questions/new` gaan en een nieuwe vraag

> maken met keuzes!

Reviewer: ```suggestion
Nu kan je naar `localhost:3000/questions/new` gaan en een nieuwe vraag met keuzes
maken!
```

}
```

> ## Leten we tot slot het bewerken van keuzes voor bestaande vragen mogelijk maken {#edit-choices-for-question}

Reviewer: ```suggestion
## Laten we tot slot het bewerken van keuzes voor bestaande vragen mogelijk maken {#edit-choices-for-question}

```

```

[`upsert`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upsert)

> is een speciale operatie die betekent: "Als dit item bestaat, update het

Reviewer: ```suggestion
is een speciale handeling die betekent: "Als dit item bestaat, update het
```

Leek mij duidelijker in deze context. Operatie doet me meer aan de ziekenhuisbehandeling denken.

Author: Misschien hier **database operatie** van maken? De term operatie wordt in deze context wel gebruikt in onze sector natuurlijk. Handeling vind ik wat vreemder.

Reviewer: Ja databaseoperatie is nog beter.  Ben niet zeker van de spatie. Beide variaties worden blijkbaar gebruikt.

Author: Google wilt "databaseoperatie" altijd veranderen naar "database operatie" 😄 Ik vind het af elkaar nog iets duidelijker om te lezen. Een samenstelling met twee klinkers tegen elkaar vind ik er vaak wat moeilijk uitzien.

dan. Zo niet, maak het dan aan". Dat is perfect in dit geval, want we
hebben de gebruiker niet verplicht om drie keuzes toe voegen bij het maken
van de vraag. Dus als de gebruiker later nog een keuze toevoegt door de
vraag te bewerken, zal het die hier worden aangemaakt.

## Conclusie {#conclusion}

🥳 Proficiat! Je hebt je eigen Blitz app gemaakt! Speel er gerust wat mee,
of deel het met je vrienden. Nu je deze tutorial hebt afgewerkt, waarom

> probeer je je stemming-app niet nog beter te maken? Je zou kunnen
> proberen:

Reviewer: ```suggestion
probeer je je stemming app niet nog beter te maken? Je zou kunnen
proberen om:
```

> - Styling toevoegen (Hint: probeer `blitz install tailwind` of

Reviewer: ```suggestion
- Styling toe te voegen (Hint: probeer `blitz install tailwind` of
```

  `blitz install chakra-ui`)

> - Meer statistieken over de stemmen laten zien
> - Live uitrollen op [Render](./deploy-render) of [Vercel](./deploy-vercel)

Reviewer: ```suggestion
- Meer statistieken over de stemmen te laten zien
- Live uit te rollen op [Render](./deploy-render) of [Vercel](./deploy-vercel)
```
